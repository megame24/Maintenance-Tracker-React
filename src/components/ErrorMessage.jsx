import React from 'react';
import { PropTypes } from 'prop-types';

const ErrorMessage = ({ errors }) => (
  errors.message
    ? (
      <div className="message error-message">
        {errors.message}
      </div>
    ) : null
);

ErrorMessage.propTypes = {
  errors: PropTypes.shape({
    statusCode: PropTypes.number
  }),
};

ErrorMessage.defaultProps = {
  errors: {},
};

export default ErrorMessage;
