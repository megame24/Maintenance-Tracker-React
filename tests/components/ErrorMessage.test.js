import React from 'react';
import { shallow } from "enzyme";
import ErrorMessage from '../../src/components/ErrorMessage';

describe('Testing ErrorMessage component', () => {
  it('should render as expected when there is an error', () => {
    const errors = {
      message: 'error'
    }
    const tree = shallow(<ErrorMessage errors={errors} />);
    expect(tree).toMatchSnapshot();
  });
  it('should return null when there is no error', () => {
    const errors = {};
    const tree = shallow(<ErrorMessage errors={errors} />);
    expect(tree).toBeNull;
  });
});
