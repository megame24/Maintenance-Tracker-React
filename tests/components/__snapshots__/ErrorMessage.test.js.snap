// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing ErrorMessage component should render as expected when there is an error 1`] = `
<div
  className="message error-message"
>
  error
</div>
`;
